CONTRIBUTING.md

```diff
@@ -59,6 +59,22 @@ During the collector release process, all `./.chloggen/*.yaml` files are transcr
 
 Alternately, copy `./.chloggen/TEMPLATE.yaml`, or just create your file from scratch.
 
+## Platform Agnostic Code
+
+In order to ensure compatibility with different operating systems, it is advised to write a code that is platform-agnostic. Below are some guidelines to follow when writing platform-agnostic code:
+
+* Avoid using platform-specific libraries, features etc. Please opt for cross-platform solutions. 
+
+* Avoid hard-coding platform specific values. Use environment variables or configuration files for storing platform specific values.
+
+* Be mindful of 
+  - Standard file systems and file paths such as forward slashes (/) instead of backward slashes (\) in Windows. 
+  - Consistent line ending formats such as Unix (LF) or Windows (CRLF).
+
+* Test your implementation thoroughly on different platforms and fix any issues. 
```

Comment:
I agree with this statement, but the Contrib CI is currently not enforcing Windows tests with every build :/ Doesn't feel right to recommend it when we aren't doing it ourselves.  @open-telemetry/collector-contrib-approvers what do you think?

Reply:
I think that when working on platform-dependent code (which I expect will continue to exist, even as part of a portable implementation) it is important to test on the relevant platforms.  That said, not all developers will have all platforms available to them and the CI matrix is important to provide assurance that we're consistently testing on supported platforms.

Reply:
@Aneurysm9 Should we keep the testing line or remove it for now?

Reply:
v0v

Really not sure what the right answer here is.  If multi-platform testing is needed it is needed, but for many PRs it won't be and the author can rely on the CI system to do it for them.

Reply:
How about asking the best effort testing ? Like below

`Test your implementation thoroughly on different platforms if possible and fix any issues.`